### ISSUE TITLE
v1.5.4 breaks `metaInfo` reactivity

### ISSUE THREAD

**Issue by kchung** [declandewet/vue-meta#269]
https://github.com/declandewet/vue-meta/issues/261 and https://github.com/declandewet/vue-meta/pull/262 introduced a breaking feature to the reactivity of `metaInfo`

For example, this component incrementally updates the `title` prop:
```html
<script>
export default {
  name: 'app',
  metaInfo() {
    return {
      title: this.title,
    };
  },
  data() {
    return {
      increment: 0,
      title: 'test',
    };
  },
  mounted() {
    setInterval(() => {
      // The title will *not* reactively update with 1.5.4
      this.title = `title-${this.increment++}`;
    }, 500);
  },
}
</script>
```

In `v1.5.4` it doesn't react to any of the changes like `v1.5.3` does.

### `v1.5.3`
![1 5 3](https://user-images.githubusercontent.com/879656/46051166-b5d12e80-c0ec-11e8-8deb-5317c6c5024d.gif)

### `v1.5.4` (this is a gif)
![1 5 4](https://user-images.githubusercontent.com/879656/46051167-b8338880-c0ec-11e8-8568-d5368dceb86b.gif)

I created a test project to demonstrate the issue:
* `v1.5.4` https://github.com/kchung/vue-meta-issue
* `v1.5.3` https://github.com/kchung/vue-meta-issue/tree/1.5.3

You can test by running `npm run serve` after running `npm install`

**Comment by yikuo123** [comment]
I encountered the same problem

**Comment by manniL** [comment]
Thanks for the detailed issue! We will look into it ☺️

**Comment by LordGuard** [comment]
same problem

**Comment by homerjam** [comment]
I've got this issue too. After upgrading a Nuxt project to v2.0 the page titles stopped updating. Downgrading has fixed the issue for the time being `npm i -S vue-meta@1.5.3`.

Having looked at the recent commit history I think it must be something to do with [this](https://github.com/declandewet/vue-meta/commit/ad4d61c38ff8e83cf4d933b961cc8875e1e22f60#diff-e64449e75a61afb51d2f85b4a353f764). My guess is that by merging the nested options the original objects/prototypes/references are lost. I think this breaks the reactivity because AFAIK reactivity is basically monkey patching set/get methods.

**Comment by manniL** [comment]
Thanks all! Resolved with `vue-meta@1.5.5`
